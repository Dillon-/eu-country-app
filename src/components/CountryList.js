import React from 'react';
import { SectionList } from 'react-native';
import PropTypes from 'prop-types';
import { CountryListCell, CountryListHeader } from 'components';

const CountryList = ({ sections }) => {
  return (
      <SectionList
        stickySectionHeadersEnabled={false}
        keyExtractor={item => item.name}
        renderItem={({ item }) => <CountryListCell country={item} /> }
        renderSectionHeader={({ section: { title } }) => <CountryListHeader title={title} /> }
        sections={sections}
      />
  )
};

CountryList.propTypes = {
  sections: PropTypes.array.isRequired,
};

export default CountryList;
