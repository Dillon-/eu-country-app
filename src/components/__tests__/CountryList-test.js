import React from 'react';
import renderer from 'react-test-renderer';
import CountryList from '../CountryList';

it('renders a CountryList', () => {
  const sections = [
    {
      title: 'North America',
      data: [
        { name: 'USA', subregion: 'North America', capital: 'Washington DC', languages: [{ name: 'English' }]},
        { name: 'Canada', subregion: 'North America', capital: 'Ottawa', languages: [{ name: 'English' }, { name: 'French' }]},
      ]
    }
  ]
  const component = renderer.create(
    <CountryList
      sections={sections}
    />
  );
  expect(component).toMatchSnapshot();
});