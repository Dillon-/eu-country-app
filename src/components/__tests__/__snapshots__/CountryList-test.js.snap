// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a CountryList 1`] = `
<RCTScrollView
  ItemSeparatorComponent={undefined}
  data={
    Array [
      Object {
        "data": Array [
          Object {
            "capital": "Washington DC",
            "languages": Array [
              Object {
                "name": "English",
              },
            ],
            "name": "USA",
            "subregion": "North America",
          },
          Object {
            "capital": "Ottawa",
            "languages": Array [
              Object {
                "name": "English",
              },
              Object {
                "name": "French",
              },
            ],
            "name": "Canada",
            "subregion": "North America",
          },
        ],
        "title": "North America",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  invertStickyHeaders={undefined}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  onViewableItemsChanged={undefined}
  renderItem={[Function]}
  renderSectionHeader={[Function]}
  scrollEventThrottle={50}
  sections={
    Array [
      Object {
        "data": Array [
          Object {
            "capital": "Washington DC",
            "languages": Array [
              Object {
                "name": "English",
              },
            ],
            "name": "USA",
            "subregion": "North America",
          },
          Object {
            "capital": "Ottawa",
            "languages": Array [
              Object {
                "name": "English",
              },
              Object {
                "name": "French",
              },
            ],
            "name": "Canada",
            "subregion": "North America",
          },
        ],
        "title": "North America",
      },
    ]
  }
  stickyHeaderIndices={Array []}
  stickySectionHeadersEnabled={false}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={undefined}
      >
        North America
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          USA
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={undefined}
        >
          North America
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={undefined}
        >
          Washington DC
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={undefined}
        >
          Lang: 
          English
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          Canada
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={undefined}
        >
          North America
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={undefined}
        >
          Ottawa
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={undefined}
        >
          Lang: 
          English, French
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    />
  </View>
</RCTScrollView>
`;
